src/box/lua/config/applier/box_cfg.lua

```diff
@@ -343,6 +344,12 @@ local function apply(config)
         box_cfg.replication = peer_uris(configdata)
     end
 
+    local replication = configdata:get('replication', {use_default = true})
+    if replication ~= nil then
+        replication['sync_timeout'] =
+            compat.box_cfg_replication_sync_timeout:is_old() and 300 or 0
+    end
```

Comment:
I've described all the details in the previous comment. Here I'll summarize problems and changes that I expect.

1. This code replaces a value that is written by a user in the YAML configuration with a default.
2. It seems, just `default = box.NULL` works correctly -- `load_cfg.lua` has a logic to determine the default depending on the compat option. We can drop this block of code from the applier.

Reply:
Removed code from `box_cfg.lua`. I also added a test that checks that `replication.sync_timeout` is changed to the default value after `config:reload()` (case from "Example 3:").